### ISSUE TITLE
Input binding fails for EventHub trigger when the binding type is List

### ISSUE THREAD

**Issue by kshyju** [Azure/azure-functions-dotnet-worker#1382]
Input binding fails💥 for EventHub trigger when the binding type is `List<T>`! See the below 2 functions. Run1 works fine, Run2 fails because `inputs` is not populated and is `null`

```csharp
  [Function("MyEventHubTriggerEnumerable")]
  public void Run1([EventHubTrigger("samples-workitems", Connection = "Con")] IEnumerable<string> inputs)
  {
      _logger.LogInformation($"IEnumerable count: {inputs?.Count()}");
      foreach (var item in inputs)
      {
          _logger.LogInformation($"{item}");
      }
  }

  [Function("MyEventHubTriggerList")]
  public void Run2([EventHubTrigger("samples-workitems", Connection = "Con")] List<string> inputs)
  {
      _logger.LogInformation($"List Length: {inputs.Count}");
      foreach (var item in inputs)
      {
          _logger.LogInformation($"{item}");
      }
  }
```

Works fine for `IEnumerable<string>`, `ICollection<string>`, `IList<string>` and `IReadOnlyList<string>`. `List<string>` is the one unhappy puppy.

When target type is `IEnumerable<string>`, `ICollection<string>`, `IList<string>` or `IReadOnlyList<string>`, The [TypeConverter ](https://github.com/Azure/azure-functions-dotnet-worker/blob/2b78e7b03503f499ae71507b2a842d1848b86fb2/src/DotNetWorker.Core/Converters/TypeConverter.cs#L9) does the conversion. But it does not work for array or List.  

![image](https://user-images.githubusercontent.com/144469/221059319-e9d42bd1-c065-483c-b4bb-e3f540923fa9.png)

When target type is `string[`], the [TypeConverter ](https://github.com/Azure/azure-functions-dotnet-worker/blob/2b78e7b03503f499ae71507b2a842d1848b86fb2/src/DotNetWorker.Core/Converters/TypeConverter.cs#L9) still returns an unhandled response, but the [ArrayConverter ](https://github.com/Azure/azure-functions-dotnet-worker/blob/2b78e7b03503f499ae71507b2a842d1848b86fb2/src/DotNetWorker.Core/Converters/ArrayConverter.cs#L12)can successfully convert.

![image](https://user-images.githubusercontent.com/144469/221059769-bcbd02a8-30a9-497e-aec5-3e18bba6d7e2.png)

In short, we do not have a converter which is able to convert when target type is `List<string>` in this example.

**Comment by mattchenderson** [comment]
Related: https://github.com/Azure/azure-functions-dotnet-worker/issues/1334
